app/main/posts/modify/post-data-editor.controller.js

```diff
@@ -301,22 +308,24 @@ function PostDataEditorController(
         return PostEditService.validatePost($scope.post, $scope.parentForm.form, $scope.tasks);
     }
 
-    function cancel(url) {
-        $scope.isLoading.state = false;
-        $scope.savingPost.saving = false;
-        /** @DEVNOTE I think we shouldn't need to check this,
-         * but in unstructured posts the lock is not available consistently.
-        **/
-        if ($scope.post.lock) {
-            PostLockEndpoint.unlock({
-                id: $scope.post.lock.id,
-                post_id: $scope.post.id
-            }).$promise.then(function (result) {
-                doChangePage(url);
-            });
-        } else {
-            doChangePage(url);
-        }
+    function cancel() {
```

Comment:
These function names are really unhelpful. We've not cancelling, we're mid way through the cancelling process. 
This should probably be something more like `unlockPost`